libsql-client/src/connection.rs

```diff
@@ -56,3 +56,93 @@ pub trait Connection {
         Ok(ret)
     }
 }
+
+/// A generic connection struct, wrapping possible backends.
+/// It's a convenience struct which allows implementing connect()
+/// with backends being passed as env parameters.
+pub enum GenericConnection {
+    #[cfg(feature = "local_backend")]
+    Local(super::local::Connection),
+    #[cfg(feature = "reqwest_backend")]
+    Reqwest(super::reqwest::Connection),
+    #[cfg(feature = "workers_backend")]
+    Workers(super::workers::Connection),
+}
+
+#[async_trait(?Send)]
+impl Connection for GenericConnection {
+    async fn batch(
+        &self,
+        stmts: impl IntoIterator<Item = impl Into<Statement>>,
+    ) -> Result<Vec<QueryResult>> {
+        match self {
+            #[cfg(feature = "local_backend")]
+            Self::Local(l) => l.batch(stmts).await,
+            #[cfg(feature = "reqwest_backend")]
+            Self::Reqwest(r) => r.batch(stmts).await,
+            #[cfg(feature = "workers_backend")]
+            Self::Workers(w) => w.batch(stmts).await,
+        }
+    }
+}
+
+/// Establishes a database connection based on environment variables
+///
+/// # Env
+/// * `LIBSQL_CLIENT_BACKEND` - one of the available backends; e.g. `reqwest`, `local`, `workers`
+/// * `LIBSQL_CLIENT_URL` - URL of the database endpoint - e.g. a https:// endpoint for remote connections (with specified credentials) or local file:/// path for a local database
+/// *
+/// # Examples
+///
+/// ```
+/// # use libsql_client::Connection;
+/// use url::Url;
+///
+/// let url  = Url::parse("https://foo:bar@localhost:8080").unwrap();
+/// let db = Connection::connect_from_url(&url).unwrap();
+/// ```
+pub fn connect() -> anyhow::Result<GenericConnection> {
+    /*
+    #[cfg(feature = "workers_backend")]
+    pub mod workers;
+
+    #[cfg(feature = "reqwest_backend")]
+    pub mod reqwest;
+
+    #[cfg(feature = "local_backend")]
+    pub mod local;
+        */
+    let url = std::env::var("LIBSQL_CLIENT_URL").map_err(|_| {
+        anyhow::anyhow!("LIBSQL_CLIENT_URL variable should point to your libSQL/sqld database")
+    })?;
+    let backend = std::env::var("LIBSQL_CLIENT_BACKEND").unwrap_or_else(|_| {
+        if url.starts_with("http") {
+            return if cfg!(feature = "reqwest_backend") {
+                "reqwest"
+            } else if cfg!(feature = "workers_backend") {
+                "workers"
+            } else {
+                "local"
```

Comment:
Will an http url work with local backend? It seems that we could just throw an exception here, no?

Reply:
if somebody decides to explicitly set the _BACKEND to local and the URL to http, they'll get an error that "https://something-something" is not a valid filesystem path - the error message is quite clear in this case, so (I hope) users would realize what's wrong